grpc/runtime/src/main/scala/io/buoyant/grpc/runtime/Stream.scala

```diff
@@ -22,8 +22,22 @@ object Stream {
     def close(): Future[Unit]
   }
 
+  private def failAndDrainQueue[T](q: AsyncQueue[Releasable[T]], e: Throwable): Unit = {
```

Comment:
Would be nice if there was a short comment here explaining why this is different from calling `q.fail(..., discard = true)` and why it's used instead.